src/NeoSharp.Core/VM/StackAccessor.cs

```diff
@@ -75,12 +81,26 @@ public byte[] PopByteArray()
 
         public T Pop<T>() where T : class
         {
-            var stackItem = _stack.Pop() as InteropStackItemBase<T>;
+            var stackItem = _stack.Pop();
 
-            using (stackItem)
+            if (stackItem is InteropStackItemBase<T> interop)
             {
-                return stackItem?.Value;
+                using (stackItem)
+                {
+                    return interop?.Value;
+                }
             }
+            else
+            {
+                // Extract base type
+
+                if (typeof(T).IsAssignableFrom(stackItem.GetType()))
+                {
+                    return (T)(object)stackItem;
```

Comment:
`using`?

Reply:
I need this for take the type, and the base object (for compare with the json) so the using should be outside of this